### ISSUE TITLE
How to get one mask per image

### ISSUE THREAD

**Issue by hanjoonwon** [facebookresearch/segment-anything#659]
I ran it with my current images and it seems to be storing too many masks for one image and is inaccurate. I want a foreground mask image so I need one mask for each image, how do I modify the code?
my image https://github.com/NVlabs/instant-ngp/tree/master/data/nerf/fox/images

**Comment by pinguo-huxiaohe** [comment]
multimask_output: bool = False can return one mask

**Comment by hanjoonwon** [comment]
> multimask_output: bool = False can return one mask

thank you :)  Will only the most accurate masks be output?

**Comment by pinguo-huxiaohe** [comment]
source code in SAM:
from segment_anything import sam_model_registry, SamPredictor
sam_checkpoint = "sam_vit_h_4b8939.pth"
model_type = "vit_h"
device = "cuda"
sam = sam_model_registry[model_type](checkpoint=sam_checkpoint)
sam.to(device=device)

predictor = SamPredictor(sam)
masks, scores, logits = predictor.predict(
    point_coords=input_point,
    point_labels=input_label,
    multimask_output=True,
)

the best single mask can be chosen by picking the one with the highest score returned in scores.
Based on my experience, if you want to use 'multimask_output=True' and get the mask you want, you should give more prompts , such as points or boxs

**Comment by pinguo-huxiaohe** [comment]
more info:https://github.com/facebookresearch/segment-anything/blob/main/notebooks/predictor_example.ipynb

**Comment by hanjoonwon** [comment]
> more info:https://github.com/facebookresearch/segment-anything/blob/main/notebooks/predictor_example.ipynb

thanks! Not directly applicable in the ubuntu anaconda environment, but should use notebooks.

**Comment by pinguo-huxiaohe** [comment]
it can use in anaconda, download the source code and weights,
"
import cv2
import numpy
from segment_anything import SamPredictor, sam_model_registry

# sam = sam_model_registry["vit_h"](checkpoint="weights/sam_vit_h_4b8939.pth")
# sam = sam_model_registry["vit_l"](checkpoint="weights/sam_vit_l_0b3195.pth")
sam = sam_model_registry["vit_b"](checkpoint="weights/sam_vit_b_01ec64.pth")
predictor = SamPredictor(sam)

image_data = cv2.imread(
    '...')
predictor.set_image(image_data)
input_point = numpy.array([[500, 375]])
input_label = numpy.array([1])
masks, scores, logits = predictor.predict(
    point_coords=input_point,
    point_labels=input_label,
    multimask_output=True,
)
"

**Comment by hanjoonwon** [comment]
> it can use in anaconda, download the source code and weights, " import cv2 import numpy from segment_anything import SamPredictor, sam_model_registry
> 
> # sam = sam_model_registry["vit_h"](checkpoint=%22weights/sam_vit_h_4b8939.pth%22)
> # sam = sam_model_registry["vit_l"](checkpoint=%22weights/sam_vit_l_0b3195.pth%22)
> sam = sam_model_registry["vit_b"](checkpoint=%22weights/sam_vit_b_01ec64.pth%22) predictor = SamPredictor(sam)
> 
> image_data = cv2.imread( '...') predictor.set_image(image_data) input_point = numpy.array([[500, 375]]) input_label = numpy.array([1]) masks, scores, logits = predictor.predict( point_coords=input_point, point_labels=input_label, multimask_output=True, ) "

thank you so much!
